packages/chatbot-server-mongodb-public/src/config.ts

```diff
@@ -129,18 +136,47 @@ export const conversations = makeMongoDbConversationsService(
   systemPrompt
 );
 
+export const createCustomConversationDataWithIpAuthUserAndOrigin: AddCustomDataFunc =
+  async (req, res) => {
+    const customData: ConversationCustomData = {};
+    if (req.cookies.auth_user) {
+      customData.authUser = req.cookies.auth_user;
+    }
+    if (req.ip) {
+      customData.ip = req.ip;
+    }
+    if (res.locals.customData.origin) {
+      customData.origin = res.locals.customData.origin;
+    }
+    logRequest({
+      reqId: getRequestId(req),
+      message: `Custom data: ${customData}`,
+    });
+    return customData;
+  };
+
+const isProduction = process.env.NODE_ENV === "production";
 export const config: AppConfig = {
   conversationsRouterConfig: {
     dataStreamer,
     llm,
     conversations,
+    middleware: [
+      requireRequestOrigin(),
+      requireValidIpAddress(),
+      cookieParser(),
+    ],
+    createConversationCustomData: !isProduction
+      ? createCustomConversationDataWithIpAuthUserAndOrigin
+      : undefined,
     generateUserPrompt,
     maxUserMessagesInConversation: 50,
     maxUserCommentLength: 500,
   },
   maxRequestTimeoutMs: 30000,
   corsOptions: {
     origin: allowedOrigins,
+    credentials: !isProduction,
```

Comment:
AFAIK this means that we allow requests to include cookies in non-prod environments.

Since CORS is kind of tricky can we add a comment here explaining this?

Also why are we allowing cookies in dev/staging but not prod?

Reply:
> Also why are we allowing cookies in dev/staging but not prod?

allow in dev/staging b/c we use to get user from cookie.

whereas we don't do this in prod b/c we don't need to. realistically, i don't think there's any threat vector here b/c the prod server isn't touching cookies. my initial thought was only allow for low lvls b/c of principle of least privilege, but it probably doesn't meaningfully matter here.

Reply:
adding comment and have it always accept credentials b/c upon research and analysis, there's no threat vector to us by allowing browser to send credentials.